support/helper.py

```diff
@@ -141,63 +160,216 @@ def dict_reverse_cumsum(d, col, sum_col):
     return data.to_dict('records')
 
 
-def aggregate_dict_by_month(data):
+def aggregate_dict_by_month(data, agg_dict):
     df = pd.DataFrame(data).set_index('date')
     df.index = pd.to_datetime(df.index)
-    df = df.groupby([df.index.year.values, df.index.month.values]).agg({'amount': 'sum', 'limit': 'max'})
+    df = df.groupby([df.index.year.values, df.index.month.values]).agg(agg_dict)
     return df
 
 
 def util_ratio(metadata, data):
     metadata['credit_card']['util_ratio'] = {}
+    metadata['credit_card']['util_ratio']['general'] = {}
+    metadata['credit_card']['util_ratio']['period'] = {}
     period = [30, 60, 90, 180, 360, 720, 1800]
     months = [-1, -2, -3, -6, -12, -24, -60]
     data['util_ratio'] = data['amount'] / data['limit']
     for p, m in zip(period, months):
-        metadata['credit_card']['util_ratio'][p] = data.iloc[m:].util_ratio.max()
+        metadata['credit_card']['util_ratio']['period'][p] = data.iloc[m:].util_ratio.max()
+    temp = data[data['util_ratio'] > 0]
+    metadata['credit_card']['util_ratio']['general']['avg_monthly_value'] = temp['util_ratio'].mean()
+    metadata['credit_card']['util_ratio']['general']['month_count'] = len(data['util_ratio'])
     return metadata
 
 
-def balances(metadata, lst, key, df=None):
-    metadata[key]['balances'] = {}
-    metadata[key]['accounts'] = {}
-    current, limit, high_balance = [], [], []
+def general(metadata, lst, k1, k2='general', df=None):
```

Comment:
Idea: we could **add docstrings** for long functions like this one that have two or more function parameters.